app/services/messages/service.go

```diff
@@ -305,6 +307,22 @@ func (ss *Service) submitEthTxTopicMessage(transferID, messageHash, ethereumTxHa
 	return ss.hederaClient.SubmitTopicConsensusMessage(ss.topicID, ethTxHashBytes)
 }
 
+func (ss *Service) awaitTransfer(transferID string) (*entity.Transfer, error) {
```

Comment:
Maybe we could add inline comment on why we do this, since this is an edge case that we discovered. Its important info imo